packages/lesswrong/components/posts/PostsPage/ImageCropPreview.tsx

```diff
@@ -0,0 +1,457 @@
+        ...leftOffsets,
+        leftFlipped: coordsLeft.flipped,
+        ...middleOffsets,
+        middleFlipped: coordsMiddle.flipped,
+        ...rightOffsets,
+        rightFlipped: coordsRight.flipped
+      };
+  
+      const { errors } = await createSplashArtCoordinateMutation({ data: splashArtData });
+      
+      if (errors) {
+        // eslint-disable-next-line no-console
+        console.error('Error(s) when saving coordinates', { errors: JSON.stringify(errors, null, 2) });
+        setShowSaveSuccess(false);
+      } else {
+        setShowSaveSuccess(true);
+      }
+    }
+    catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('Error saving coordinates', error);
+      setShowSaveSuccess(false); // Set failure state
+    }
+  }, [selectedImageInfo, createSplashArtCoordinateMutation, cachedBoxCoordinates, imgRef]);
+
+  if (!selectedImageInfo) {
+    return null;
+  }
+
+  const moveableBoxStyle = {
+    left: boxCoordinates.x,
+    top: boxCoordinates.y,
+    backgroundPosition: `-${boxCoordinates.x}px -${boxCoordinates.y}px`, // Set the background position based on boxPosition
+    backgroundSize: `${windowSize.width}px auto`, // Ensure the background image covers the entire screen     
+    width: boxCoordinates.width,
+    height: boxCoordinates.height,             
+  };
+
+  const boxSubContainers = {
+    display: 'flex',
+    justifyContent: 'space-around',
+  };
+
+  const showSaveAllButton = ( 
+    cachedBoxCoordinates[selectedImageInfo._id] && 
+    cachedBoxCoordinates[selectedImageInfo._id]["left"] && 
+    cachedBoxCoordinates[selectedImageInfo._id]["middle"] && 
+    cachedBoxCoordinates[selectedImageInfo._id]["right"]
+  );
+
+  const previewSubsetProps = {
+    boxCoordinates,
+    selectedImageInfo,
+    selectedBox,
+    setSelectedBox,
+    cachedBoxCoordinates,
+    setCachedBoxCoordinates,
+    flipped
+  };
+
+  return (
+    <>
+      <button className={classes.button} onClick={toggleBoxVisibility}>Show Box</button>
+      {isBoxVisible && selectedImageInfo && selectedImageInfo._id && (
+        <div className={classes.moveableBox}
+          style={moveableBoxStyle}
+          onMouseDown={handleMouseDown}>
+          <div style={boxSubContainers}>
+            <ImagePreviewSubset subBoxPosition='left' {...previewSubsetProps} />
+            <ImagePreviewSubset subBoxPosition='middle' {...previewSubsetProps} />
+            <ImagePreviewSubset subBoxPosition='right' {...previewSubsetProps} />
+          </div>
+          <div className={classes.saveAllCoordinates} onClick={saveAllCoordinates} style={{backgroundColor: showSaveAllButton ? 'rgba(0, 255, 0, 0.7)' : 'rgba(255, 0, 0, 0.3)'}}>
+            {!showSaveAllButton
+              ? (<div> Must set all placements before you can save them </div>)
+              : loading
+                ? <div>Saving all placements...</div> 
+                : <div onClick={saveAllCoordinates}>{`Save all placements`}</div>
```

Comment:
No nested ternaries, plz - extract to some more legible logic in the component that assigns to a variable that gets used here